src/BPEssentials/ExtendedPlayer/ExtendedPlayer.cs

```diff
@@ -38,5 +41,34 @@ public enum Chat
             Global,
             StaffChat
         }
+
+        public class LastLocation
+        {
+            public Vector3 Position { get; private set; }
+            public Quaternion Rotation { get; private set; }
+            public int PlaceIndex { get; private set; }
+            public void Update(Vector3 position, Quaternion rotation, int index)
+            {
+                Position = position;
+                Rotation = rotation;
+                PlaceIndex = index;
+            }
+            public void Update(ShPlayer player)
+            {
+                Update(player.GetPosition(), player.GetRotation(), player.GetPlaceIndex());
+            }
+            public bool HasPositionSet()
+            {
+                return Position != default(Vector3) && Rotation != default(Quaternion);
+            }
+            public LastLocation(Vector3 position, Quaternion rotation, int index)
+            {
+                Update(position, rotation, index);
+            }
+            public LastLocation()
+            {
+                
+            }
+        }
```

Comment:
Please add new lines in between methods and properties. This is pretty hard to suggest using github, so just gonna comment it here.